### ISSUE TITLE
App Scrolling Issue

### ISSUE THREAD

**Issue by wasowerby** [xamarin/xamarin-android#1480]
### Steps to Reproduce

[Skylog.SkylogPro.zip](https://github.com/xamarin/xamarin-android/files/1851414/Skylog.SkylogPro.zip)
1. Run the attached zipped Skylog.SkylogPro.apk and select 'Flight Records'
2. Touch one of the records and display the 'Flight Records' screen.
3. Scroll the (Resco Toolkit's DetailView) fields up and down a few times.
4. Select 'Cancel' button and observe 'Unfortunately, Skylog Pro has stopped' message.
5. Select 'Ok' and observe we are back at the 'Main Menu' screen NOT the 'Flight Records' screen.
6. Repeat 1 - 2 and omit the scrolling in 3, and note that in 4, we are now back, correctly, at the 'Flight Records' screen.
7. Note that this behaviour does not occur if Xamarin 4.1.2.18 is used (in VS2015) but does with 4.2 onwards.

### Expected Behavior
After cancelling the detailed 'Flight Record' screen, the app should always return to the 'Flight Records' screen irrespective of any scrolling activity.
### Actual Behavior
As detailed above.
### Version Information
Microsoft Visual Studio Community 2017 
Version 15.6.4
VisualStudio.15.Release/15.6.4+27428.2015
Microsoft .NET Framework
Version 4.7.02556

Installed Version: Community

Visual C++ 2017   00369-60000-00001-AA324
Microsoft Visual C++ 2017

Visual F# Tools 10.1 for F# 4.1   00369-60000-00001-AA324
Microsoft Visual F# Tools 10.1 for F# 4.1

Application Insights Tools for Visual Studio Package   8.11.10212.1
Application Insights Tools for Visual Studio

ASP.NET and Web Tools 2017   15.0.40314.0
ASP.NET and Web Tools 2017

ASP.NET Core Razor Language Services   1.0
Provides languages services for ASP.NET Core Razor.

Azure App Service Tools v3.0.0   15.0.40215.0
Azure App Service Tools v3.0.0

C# Tools   2.7.0-beta3-62715-05. Commit Hash: db02128e6e3c4bdfc93e6ec425ac9162b4d4fe80
C# components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Common Azure Tools   1.10
Provides common services for use by Azure Mobile Services and Microsoft Azure Tools.

JavaScript Language Service   2.0
JavaScript Language Service

JavaScript Project System   2.0
JavaScript Project System

Merq   1.1.17-rc (cba4571)
Command Bus, Event Stream and Async Manager for Visual Studio extensions.

Microsoft Continuous Delivery Tools for Visual Studio   0.3
Simplifying the configuration of continuous build integration and continuous build delivery from within the Visual Studio IDE.

Microsoft JVM Debugger   1.0
Provides support for connecting the Visual Studio debugger to JDWP compatible Java Virtual Machines

Microsoft MI-Based Debugger   1.0
Provides support for connecting Visual Studio to MI compatible debuggers

Microsoft Visual C++ Wizards   1.0
Microsoft Visual C++ Wizards

Microsoft Visual Studio VC Package   1.0
Microsoft Visual Studio VC Package

Mono Debugging for Visual Studio   4.9.11-pre (71eb098)
Support for debugging Mono processes with Visual Studio.

NuGet Package Manager   4.6.0
NuGet Package Manager in Visual Studio. For more information about NuGet, visit http://docs.nuget.org/.

ProjectServicesPackage Extension   1.0
ProjectServicesPackage Visual Studio Extension Detailed Info

SQL Server Data Tools   15.1.61801.210
Microsoft SQL Server Data Tools

Test Adapter for Boost.Test   1.0
Enables Visual Studio's testing tools with unit tests written for Boost.Test.  The use terms and Third Party Notices are available in the extension installation directory.

Test Adapter for Google Test   1.0
Enables Visual Studio's testing tools with unit tests written for Google Test.  The use terms and Third Party Notices are available in the extension installation directory.

TypeScript Tools   15.6.20202.3
TypeScript Tools for Microsoft Visual Studio

Visual Basic Tools   2.7.0-beta3-62715-05. Commit Hash: db02128e6e3c4bdfc93e6ec425ac9162b4d4fe80
Visual Basic components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Visual Studio Code Debug Adapter Host Package   1.0
Interop layer for hosting Visual Studio Code debug adapters in Visual Studio

VisualStudio.Mac   1.0
Mac Extension for Visual Studio

Xamarin   4.9.0.752 (0faa808ee)
Visual Studio extension to enable development for Xamarin.iOS and Xamarin.Android.

Xamarin Designer   4.10.58 (cee1369d0)
Visual Studio extension to enable Xamarin Designer tools in Visual Studio.

Xamarin.Android SDK   8.2.0.16 (HEAD/a78295902)
Xamarin.Android Reference Assemblies and MSBuild support.

Xamarin.iOS and Xamarin.Mac SDK   11.8.1.28 (3979d08)
Xamarin.iOS and Xamarin.Mac Reference Assemblies and MSBuild support.

### Log File

```
D/Mono    ( 6911): AOT: image 'Mono.Data.Sqlite.dll.so' not found: dlopen failed: library "/data/data/Skylog.SkylogPro/lib/libaot-Mono.Data.Sqlite.dll.so" not found
D/Mono    ( 6911): AOT: image '/usr/local/lib/mono/aot-cache/arm/Mono.Data.Sqlite.dll.so' not found: dlopen failed: library "/data/data/Skylog.SkylogPro/lib/libaot-Mono.Data.Sqlite.dll.so" not found
D/Mono    ( 6911): Config attempting to parse: 'Mono.Data.Sqlite.dll.config'.
D/Mono    ( 6911): Config attempting to parse: '/usr/local/etc/mono/assemblies/Mono.Data.Sqlite/Mono.Data.Sqlite.config'.
D/Mono    ( 6911): Assembly Ref addref Skylog Pro[0x5b2dc7a8] -> Mono.Data.Sqlite[0x60c09038]: 2
D/Mono    ( 6911): Image addref System.Data[0x60d9c358] -> System.Data.dll[0x60c04828]: 2
D/Mono    ( 6911): Prepared to set up assembly 'System.Data' (System.Data.dll)
D/Mono    ( 6911): Assembly System.Data[0x60d9c358] added to domain RootDomain, ref_count=1
D/Mono    ( 6911): AOT: image 'System.Data.dll.so' not found: dlopen failed: library "/data/data/Skylog.SkylogPro/lib/libaot-System.Data.dll.so" not found
D/Mono    ( 6911): AOT: image '/usr/local/lib/mono/aot-cache/arm/System.Data.dll.so' not found: dlopen failed: library "/data/data/Skylog.SkylogPro/lib/libaot-System.Data.dll.so" not found
D/Mono    ( 6911): Config attempting to parse: 'System.Data.dll.config'.
D/Mono    ( 6911): Config attempting to parse: '/usr/local/etc/mono/assemblies/System.Data/System.Data.config'.
D/Mono    ( 6911): Assembly Ref addref Mono.Data.Sqlite[0x60c09038] -> System.Data[0x60d9c358]: 2
D/Mono    ( 6911): Image addref System[0x60c19d18] -> System.dll[0x60c18f58]: 2
D/Mono    ( 6911): Prepared to set up assembly 'System' (System.dll)
D/Mono    ( 6911): Assembly System[0x60c19d18] added to domain RootDomain, ref_count=1
D/Mono    ( 6911): AOT: image 'System.dll.so' not found: dlopen failed: library "/data/data/Skylog.SkylogPro/lib/libaot-System.dll.so" not found
D/Mono    ( 6911): AOT: image '/usr/local/lib/mono/aot-cache/arm/System.dll.so' not found: dlopen failed: library "/data/data/Skylog.SkylogPro/lib/libaot-System.dll.so" not found
D/Mono    ( 6911): Config attempting to parse: 'System.dll.config'.
D/Mono    ( 6911): Config attempting to parse: '/usr/local/etc/mono/assemblies/System/System.config'.
D/Mono    ( 6911): Assembly Ref addref System.Data[0x60d9c358] -> System[0x60c19d18]: 2
D/Mono    ( 6911): Assembly Ref addref System[0x60c19d18] -> mscorlib[0x5a775b48]: 12
D/Mono    ( 6911): Assembly Ref addref System.Data[0x60d9c358] -> mscorlib[0x5a775b48]: 13
D/Mono    ( 6911): Assembly Ref addref Mono.Data.Sqlite[0x60c09038] -> mscorlib[0x5a775b48]: 14
D/Mono    ( 6911): Assembly Ref addref Skylog Pro[0x5b2dc7a8] -> mscorlib[0x5a775b48]: 15
D/Mono    ( 6911): Assembly Ref addref Skylog Pro[0x5b2dc7a8] -> Resco.Controls[0x5b24b5f0]: 2
D/Mono    ( 6911): Assembly Ref addref Resco.Controls[0x5b24b5f0] -> mscorlib[0x5a775b48]: 16
D/Mono    ( 6911): Assembly Ref addref Resco.Controls[0x5b24b5f0] -> Mono.Android[0x5b2e1450]: 3
D/Mono    ( 6911): Assembly Ref addref Resco.Controls[0x5b24b5f0] -> System[0x60c19d18]: 3
D/Mono    ( 6911): DllImport searching in: '__Internal' ('(null)').
D/Mono    ( 6911): Searching for 'java_interop_jnienv_call_nonvirtual_object_method_a'.
D/Mono    ( 6911): Probing 'java_interop_jnienv_call_nonvirtual_object_method_a'.
D/Mono    ( 6911): Found as 'java_interop_jnienv_call_nonvirtual_object_method_a'.
D/Mono    ( 6911): DllImport searching in: '__Internal' ('(null)').
D/Mono    ( 6911): Searching for 'java_interop_jnienv_new_string'.
D/Mono    ( 6911): Probing 'java_interop_jnienv_new_string'.
D/Mono    ( 6911): Found as 'java_interop_jnienv_new_string'.
D/Mono    ( 6911): DllImport searching in: '__Internal' ('(null)').
D/Mono    ( 6911): Searching for 'java_interop_jnienv_call_void_method_a'.
D/Mono    ( 6911): Probing 'java_interop_jnienv_call_void_method_a'.
D/Mono    ( 6911): Found as 'java_interop_jnienv_call_void_method_a'.
D/Mono    ( 6911): DllImport searching in: '__Internal' ('(null)').
D/Mono    ( 6911): Searching for 'java_interop_jnienv_get_field_id'.
D/Mono    ( 6911): Probing 'java_interop_jnienv_get_field_id'.
D/Mono    ( 6911): Found as 'java_interop_jnienv_get_field_id'.
D/Mono    ( 6911): DllImport searching in: '__Internal' ('(null)').
D/Mono    ( 6911): Searching for 'java_interop_jnienv_get_float_field'.
D/Mono    ( 6911): Probing 'java_interop_jnienv_get_float_field'.
D/Mono    ( 6911): Found as 'java_interop_jnienv_get_float_field'.
D/Mono    ( 6911): DllImport searching in: '__Internal' ('(null)').
D/Mono    ( 6911): Searching for 'java_interop_jnienv_set_int_field'.
D/Mono    ( 6911): Probing 'java_interop_jnienv_set_int_field'.
D/Mono    ( 6911): Found as 'java_interop_jnienv_set_int_field'.
D/Mono    ( 6911): DllImport searching in: '__Internal' ('(null)').
D/Mono    ( 6911): Searching for 'java_interop_jnienv_new_local_ref'.
D/Mono    ( 6911): Probing 'java_interop_jnienv_new_local_ref'.
D/Mono    ( 6911): Found as 'java_interop_jnienv_new_local_ref'.
D/Mono    ( 6911): DllImport searching in: '__Internal' ('(null)').
D/Mono    ( 6911): Searching for 'java_interop_jnienv_new_object_array'.
D/Mono    ( 6911): Probing 'java_interop_jnienv_new_object_array'.
D/Mono    ( 6911): Found as 'java_interop_jnienv_new_object_array'.
D/Mono    ( 6911): Image addref System.Transactions[0x608f8b78] -> System.Transactions.dll[0x608f8490]: 2
D/Mono    ( 6911): Prepared to set up assembly 'System.Transactions' (System.Transactions.dll)
D/Mono    ( 6911): Assembly System.Transactions[0x608f8b78] added to domain RootDomain, ref_count=1
D/Mono    ( 6911): AOT: image 'System.Transactions.dll.so' not found: dlopen failed: library "/data/data/Skylog.SkylogPro/lib/libaot-System.Transactions.dll.so" not found
D/Mono    ( 6911): AOT: image '/usr/local/lib/mono/aot-cache/arm/System.Transactions.dll.so' not found: dlopen failed: library "/data/data/Skylog.SkylogPro/lib/libaot-System.Transactions.dll.so" not found
D/Mono    ( 6911): Config attempting to parse: 'System.Transactions.dll.config'.
D/Mono    ( 6911): Config attempting to parse: '/usr/local/etc/mono/assemblies/System.Transactions/System.Transactions.config'.
D/Mono    ( 6911): Assembly Ref addref Mono.Data.Sqlite[0x60c09038] -> System.Transactions[0x608f8b78]: 2
D/Mono    ( 6911): Image addref System.Xml[0x60cfa768] -> System.Xml.dll[0x60cf9e18]: 2
D/Mono    ( 6911): Prepared to set up assembly 'System.Xml' (System.Xml.dll)
D/Mono    ( 6911): Assembly System.Xml[0x60cfa768] added to domain RootDomain, ref_count=1
D/Mono    ( 6911): AOT: image 'System.Xml.dll.so' not found: dlopen failed: library "/data/data/Skylog.SkylogPro/lib/libaot-System.Xml.dll.so" not found
D/Mono    ( 6911): AOT: image '/usr/local/lib/mono/aot-cache/arm/System.Xml.dll.so' not found: dlopen failed: library "/data/data/Skylog.SkylogPro/lib/libaot-System.Xml.dll.so" not found
D/Mono    ( 6911): Config attempting to parse: 'System.Xml.dll.config'.
D/Mono    ( 6911): Config attempting to parse: '/usr/local/etc/mono/assemblies/System.Xml/System.Xml.config'.
D/Mono    ( 6911): Assembly Ref addref System.Data[0x60d9c358] -> System.Xml[0x60cfa768]: 2
D/Mono    ( 6911): Assembly Ref addref System.Data[0x60d9c358] -> System.Transactions[0x608f8b78]: 3
D/Mono    ( 6911): Assembly Ref addref System.Transactions[0x608f8b78] -> mscorlib[0x5a775b48]: 17
D/Mono    ( 6911): Assembly Ref addref Skylog Pro[0x5b2dc7a8] -> System.Data[0x60d9c358]: 3
D/Mono    ( 6911): Assembly Ref addref Mono.Data.Sqlite[0x60c09038] -> System[0x60c19d18]: 4
D/Mono    ( 6911): DllImport attempting to load: 'libsqlite3_xamarin.so'.
W/linker  ( 6911): libsqlite3_xamarin.so: unused DT entry: type 0x6ffffffe arg 0x4124
W/linker  ( 6911): libsqlite3_xamarin.so: unused DT entry: type 0x6fffffff arg 0x1
D/Mono    ( 6911): DllImport loaded library './libsqlite3_xamarin.so'.
D/Mono    ( 6911): DllImport searching in: 'libsqlite3_xamarin.so' ('./libsqlite3_xamarin.so').
D/Mono    ( 6911): Searching for 'sqlite3_open_v2'.
D/Mono    ( 6911): Probing 'sqlite3_open_v2'.
D/Mono    ( 6911): Found as 'sqlite3_open_v2'.
D/Mono    ( 6911): DllImport searching in: 'libsqlite3_xamarin.so' ('./libsqlite3_xamarin.so').
D/Mono    ( 6911): Searching for 'sqlite3_open'.
D/Mono    ( 6911): Probing 'sqlite3_open'.
D/Mono    ( 6911): Found as 'sqlite3_open'.
D/Mono    ( 6911): DllImport searching in: 'libsqlite3_xamarin.so' ('./libsqlite3_xamarin.so').
D/Mono    ( 6911): Searching for 'sqlite3_libversion_number'.
D/Mono    ( 6911): Probing 'sqlite3_libversion_number'.
D/Mono    ( 6911): Found as 'sqlite3_libversion_number'.
D/Mono    ( 6911): Assembly Ref addref Skylog Pro[0x5b2dc7a8] -> System[0x60c19d18]: 5
D/Mono    ( 6911): DllImport searching in: 'libsqlite3_xamarin.so' ('./libsqlite3_xamarin.so').
D/Mono    ( 6911): Searching for 'sqlite3_busy_timeout'.
D/Mono    ( 6911): Probing 'sqlite3_busy_timeout'.
D/Mono    ( 6911): Found as 'sqlite3_busy_timeout'.
D/Mono    ( 6911): DllImport searching in: 'libsqlite3_xamarin.so' ('./libsqlite3_xamarin.so').
D/Mono    ( 6911): Searching for 'sqlite3_prepare'.
D/Mono    ( 6911): Probing 'sqlite3_prepare'.
D/Mono    ( 6911): Found as 'sqlite3_prepare'.
D/Mono    ( 6911): DllImport searching in: 'libsqlite3_xamarin.so' ('./libsqlite3_xamarin.so').
D/Mono    ( 6911): Searching for 'sqlite3_bind_parameter_count'.
D/Mono    ( 6911): Probing 'sqlite3_bind_parameter_count'.
D/Mono    ( 6911): Found as 'sqlite3_bind_parameter_count'.
D/Mono    ( 6911): DllImport searching in: 'libsqlite3_xamarin.so' ('./libsqlite3_xamarin.so').
D/Mono    ( 6911): Searching for 'sqlite3_column_count'.
D/Mono    ( 6911): Probing 'sqlite3_column_count'.
D/Mono    ( 6911): Found as 'sqlite3_column_count'.
D/Mono    ( 6911): DllImport searching in: 'libsqlite3_xamarin.so' ('./libsqlite3_xamarin.so').
D/Mono    ( 6911): Searching for 'sqlite3_step'.
D/Mono    ( 6911): Probing 'sqlite3_step'.
D/Mono    ( 6911): Found as 'sqlite3_step'.
D/Mono    ( 6911): DllImport searching in: 'libsqlite3_xamarin.so' ('./libsqlite3_xamarin.so').
D/Mono    ( 6911): Searching for 'sqlite3_changes'.
D/Mono    ( 6911): Probing 'sqlite3_changes'.
D/Mono    ( 6911): Found as 'sqlite3_changes'.
D/Mono    ( 6911): DllImport searching in: 'libsqlite3_xamarin.so' ('./libsqlite3_xamarin.so').
D/Mono    ( 6911): Searching for 'sqlite3_reset'.
D/Mono    ( 6911): Probing 'sqlite3_reset'.
D/Mono    ( 6911): Found as 'sqlite3_reset'.
D/Mono    ( 6911): DllImport searching in: 'libsqlite3_xamarin.so' ('./libsqlite3_xamarin.so').
D/Mono    ( 6911): Searching for 'sqlite3_finalize'.
D/Mono    ( 6911): Probing 'sqlite3_finalize'.
D/Mono    ( 6911): Found as 'sqlite3_finalize'.
D/Mono    ( 6911): DllImport searching in: 'libsqlite3_xamarin.so' ('./libsqlite3_xamarin.so').
D/Mono    ( 6911): Searching for 'sqlite3_column_name'.
D/Mono    ( 6911): Probing 'sqlite3_column_name'.
D/Mono    ( 6911): Found as 'sqlite3_column_name'.
D/Mono    ( 6911): DllImport searching in: 'libsqlite3_xamarin.so' ('./libsqlite3_xamarin.so').
D/Mono    ( 6911): Searching for 'sqlite3_column_decltype'.
D/Mono    ( 6911): Probing 'sqlite3_column_decltype'.
D/Mono    ( 6911): Found as 'sqlite3_column_decltype'.
D/Mono    ( 6911): DllImport searching in: 'libsqlite3_xamarin.so' ('./libsqlite3_xamarin.so').
D/Mono    ( 6911): Searching for 'sqlite3_column_type'.
D/Mono    ( 6911): Probing 'sqlite3_column_type'.
D/Mono    ( 6911): Found as 'sqlite3_column_type'.
D/Mono    ( 6911): DllImport searching in: 'libsqlite3_xamarin.so' ('./libsqlite3_xamarin.so').
D/Mono    ( 6911): Searching for 'sqlite3_column_text'.
D/Mono    ( 6911): Probing 'sqlite3_column_text'.
D/Mono    ( 6911): Found as 'sqlite3_column_text'.
D/Mono    ( 6911): DllImport searching in: 'libsqlite3_xamarin.so' ('./libsqlite3_xamarin.so').
D/Mono    ( 6911): Searching for 'sqlite3_column_int64'.
D/Mono    ( 6911): Probing 'sqlite3_column_int64'.
D/Mono    ( 6911): Found as 'sqlite3_column_int64'.
D/Mono    ( 6911): DllImport searching in: '__Internal' ('(null)').
D/Mono    ( 6911): Searching for 'java_interop_jnienv_call_boolean_method'.
D/Mono    ( 6911): Probing 'java_interop_jnienv_call_boolean_method'.
D/Mono    ( 6911): Found as 'java_interop_jnienv_call_boolean_method'.
D/Mono    ( 6911): DllImport searching in: '__Internal' ('(null)').
D/Mono    ( 6911): Searching for 'java_interop_jnienv_call_boolean_method_a'.
D/Mono    ( 6911): Probing 'java_interop_jnienv_call_boolean_method_a'.
D/Mono    ( 6911): Found as 'java_interop_jnienv_call_boolean_method_a'.
D/Mono    ( 6911): Assembly Ref addref Mono.Android[0x5b2e1450] -> System.Xml[0x60cfa768]: 3
D/Mono    ( 6911): Assembly Ref addref System.Xml[0x60cfa768] -> mscorlib[0x5a775b48]: 18
D/Mono    ( 6911): DllImport searching in: '__Internal' ('(null)').
D/Mono    ( 6911): Searching for 'java_interop_jnienv_new_int_array'.
D/Mono    ( 6911): Probing 'java_interop_jnienv_new_int_array'.
D/Mono    ( 6911): Found as 'java_interop_jnienv_new_int_array'.
D/Mono    ( 6911): DllImport searching in: '__Internal' ('(null)').
D/Mono    ( 6911): Searching for 'java_interop_jnienv_set_int_array_region'.
D/Mono    ( 6911): Probing 'java_interop_jnienv_set_int_array_region'.
D/Mono    ( 6911): Found as 'java_interop_jnienv_set_int_array_region'.
D/Mono    ( 6911): DllImport searching in: '__Internal' ('(null)').
D/Mono    ( 6911): Searching for 'java_interop_jnienv_get_int_array_region'.
D/Mono    ( 6911): Probing 'java_interop_jnienv_get_int_array_region'.
D/Mono    ( 6911): Found as 'java_interop_jnienv_get_int_array_region'.
D/Mono    ( 6911): DllImport searching in: '__Internal' ('(null)').
D/Mono    ( 6911): Searching for 'java_interop_jnienv_call_int_method_a'.
D/Mono    ( 6911): Probing 'java_interop_jnienv_call_int_method_a'.
D/Mono    ( 6911): Found as 'java_interop_jnienv_call_int_method_a'.
D/Mono    ( 6911): DllImport searching in: '__Internal' ('(null)').
D/Mono    ( 6911): Searching for 'java_interop_jnienv_get_static_field_id'.
D/Mono    ( 6911): Probing 'java_interop_jnienv_get_static_field_id'.
D/Mono    ( 6911): Found as 'java_interop_jnienv_get_static_field_id'.
D/Mono    ( 6911): DllImport searching in: '__Internal' ('(null)').
D/Mono    ( 6911): Searching for 'java_interop_jnienv_get_static_int_field'.
D/Mono    ( 6911): Probing 'java_interop_jnienv_get_static_int_field'.
D/Mono    ( 6911): Found as 'java_interop_jnienv_get_static_int_field'.
V/WindowManager(26174): Adding window Window{31f44204 u0 Skylog.SkylogPro/md5c8118ad8ef4feffccb2f16d0903c9457.ActivityMain} at 2 of 10 (before Window{1ecf60d3 u0 Starting Skylog.SkylogPro})
I/ActivityManager(26174): Displayed Skylog.SkylogPro/md5c8118ad8ef4feffccb2f16d0903c9457.ActivityMain: +2s763ms
W/InputMethodManagerService(26174): Got RemoteException sending setActive(false) notification to pid 12252 uid 10165
I/GoogleInputMethod(26635): onFinishInput() : Dummy InputConnection bound
I/GoogleInputMethod(26635): onStartInput() : Dummy InputConnection bound
V/ActivityManager(26174): Display changed displayId=0
I/ActivityManager(26174): Config changes=480 {1.0 234mcc?mnc en_GB ?layoutDir sw600dp w961dp h528dp 213dpi lrg land finger -keyb/v/h -nav/h s.28}
I/InputReader(26174): Reconfiguring input devices.  changes=0x00000004
I/InputReader(26174): Device reconfigured: id=2, name='elan-touchscreen', size 800x1280, orientation 3, mode 1, display id 0
V/WindowManager(26174): Adding window Window{3c8843a5 u0 Skylog.SkylogPro/md5c8118ad8ef4feffccb2f16d0903c9457.ActivityMain} at 2 of 8 (after Window{1f90a6a7 u0 com.android.launcher/com.android.launcher2.Launcher})
I/EmojiSearchExtension(26635): onDestroy()
I/EmojiSearchExtension(26635): onDeactivate()
I/WindowManager(26174): Screen frozen for +749ms due to Window{1ecdf76d u0 NavigationBar}
I/GoogleInputMethod(26635): onStartInput() : Dummy InputConnection bound
I/GoogleInputMethod(26635): onFinishInput() : Dummy InputConnection bound
I/GoogleInputMethod(26635): onStartInput() : Dummy InputConnection bound
E/AndroidIME(26635): bna: name should not be empty
I/art     (26174): Explicit concurrent mark sweep GC freed 39924(2MB) AllocSpace objects, 30(1201KB) LOS objects, 33% free, 25MB/38MB, paused 1.938ms total 249.095ms
I/art     (26653): Background partial concurrent mark sweep GC freed 162947(7MB) AllocSpace objects, 4(704KB) LOS objects, 39% free, 11MB/19MB, paused 3.097ms total 164.728ms
E/WifiStateMachine(26174): WifiStateMachine CMD_START_SCAN source -2 txSuccessRate=6.01 rxSuccessRate=14.66 targetRoamBSSID=92:3b:ad:2d:e9:94 RSSI=-26
E/WifiStateMachine(26174): WifiStateMachine CMD_START_SCAN source -2 txSuccessRate=0.02 rxSuccessRate=10.59 targetRoamBSSID=92:3b:ad:2d:e9:94 RSSI=-27
E/WifiStateMachine(26174): WifiStateMachine starting scan for "ORBI"WPA_PSK with 2472
E/WifiStateMachine(26174): WifiStateMachine CMD_START_SCAN source -2 txSuccessRate=0.09 rxSuccessRate=9.12 targetRoamBSSID=92:3b:ad:2d:e9:94 RSSI=-24
E/WifiStateMachine(26174): WifiStateMachine starting scan for "ORBI"WPA_PSK with 2472
```

**Comment by wasowerby** [comment]
Updated Skylog.SkylogPro.zip attached.
[Skylog.SkylogPro.zip](https://github.com/xamarin/xamarin-android/files/1852207/Skylog.SkylogPro.zip)

**Comment by wasowerby** [comment]
Given my long standing Xamarin subscription, I'm disappointed that there is no response after 2 weeks!

**Comment by alanjclark** [comment]
Can you please advise what API level you're seeing this with, and whether this is a physical device or emulator?  Do you have a project you can share that exhibits the issue?
At present, I'm seeing a fatal exception when tapping the back button and the app is re-opened and lands at the main menu each time, which I would expect.

**Comment by wasowerby** [comment]
Hi,

 

Thanks for the response. I see the issue when building with Target Marshmallow (api Level 23) and Xamarin 4.9.0.752 (actually when using Xamarin 4.2 onwards). The Skylog.SkylogPro.zip attached to the issue demonstrates – on my Nexus 7 Android Version 5.1.1 tablet – the problem. This apk was built with Xamarin 4.9.0.752 (Target Marshmallow) in VS2017. 

 

If I select the Main Menu>Flight Records screen and tap the 3rd record dated 15/04/2018 (UK date format dd/mm/yyyy) and then scroll the resulting detailed record’s fields up and down a few times and then tap the ‘Cancel’ button on the toolbar, I get the 'Unfortunately, Skylog Pro has stopped' message. On clearing that message, the screen returns to the Main Menu screen as it should of course. However normally,  after tapping the ‘Cancel’ button, the ‘Flight Records’ screen (with the 3 records) should be re-presented, and we should not get a crash with the  'Unfortunately, Skylog Pro has stopped' message. Building the project again using Xamarin 4.1.2.18 in VS2015 (and still Targetting Marshmallow) instead, fixes the problem. 

 

Basically, building with Xamarin 4.1.2.18 allows endless scrolling of a flight record’ fields whereas using Xamarin 4.9.0.752 causes a crash when that is done.

 

Hope this clarifies the issue? I can send you an apk built with Xamarin 4.1.2.18 if you wish? This will demonstrate that this problem doesn’t occur with that build.

**Comment by wasowerby** [comment]
Hope the additional info I gave will help to reproduce the issue? One further thing: If you **don't** scroll the 'Record' fields, the crash does not occur. It's the scrolling which causes the crash.

**Comment by wasowerby** [comment]
Any progress on reproducing this? Do you need any further info?

**Comment by wasowerby** [comment]
Al: Any progress on this issue?

**Comment by alanjclark** [comment]
Sorry - I misunderstood the scenario originally.  When run `adb logcat` on when the app throws the exception, I see the following:

`android.runtime.JavaProxyThrowable: System.ObjectDisposedException: Cannot access a disposed object.`

I note that we had a known issue in Xamarin.Android 7.0 (included in Xamarin.VisualStudio 4.2):
https://developer.xamarin.com/releases/android/xamarin.android_7/xamarin.android_7.0/#Known_Issues

My first assumption is that you are hitting this known issue.  However, I cannot rule out at this stage that it could be a Resco Toolkit bug. (I'm not familiar with Resco Toolkit - is this actively being developed?)

If you try the fix and are still having issues I would be interested to see a sample solution that exhibits the problem.

**Comment by wasowerby** [comment]
Hi Al, I think you're right. Adding an environmental file with:

MONO_GC_PARAMS=bridge-implementation=old

seems to have fixed the problem. Carried out a large number of 'Record' field swipes and the crash has not occurred - using Xamarin 4.9.0.752.

Many thanks for your help with this. BTW, the Resco Toolkit is no longer supported by Resco. They want $2500 for the source code! Great!

Regards...Andrew

**Comment by wasowerby** [comment]
MONO_GC_PARAMS=bridge-implementation=new
also fixes the problem.

**Comment by jpobst** [comment]
I suspect this issue is obsolete as it has not had more reports in many years.
